main.tf

```diff
@@ -51,42 +37,28 @@ resource "aws_internet_gateway" "main" {
 }
 
 # create Elastic Ip address for NAT
-resource "aws_eip" "nat0" {
-  vpc      = true
-  tags = {
-    Name = "nat0"
-  }
-}
+resource "aws_eip" "nat" {
+  count = length(var.private_cidr)
 
-# create Elastic Ipaddress for NAT
-resource "aws_eip" "nat1" {
-  vpc      = true
-  tags = {
-    Name = "nat1"
-  }
-}
-
-resource "aws_nat_gateway" "public0" {
-  allocation_id = aws_eip.nat0.id
-  subnet_id     = aws_subnet.public0.id
+  vpc = true
 
   tags = {
-    Name = "public0"
+    Name = "${var.area_code}-nat${count.index}"
   }
-
 }
 
-resource "aws_nat_gateway" "public1" {
-  allocation_id = aws_eip.nat1.id
-  subnet_id     = aws_subnet.public1.id
+resource "aws_nat_gateway" "public" {
+  count = length(var.private_cidr)
+  allocation_id = aws_eip.nat[count.index].id
```

Comment:
For blocks that contain both arguments and "meta-arguments" (as defined by the Terraform language semantics), list meta-arguments first and separate them from other arguments with one blank line. Place meta-argument blocks last and separate them from other blocks with one blank line.

https://developer.hashicorp.com/terraform/language/syntax/style